### ISSUE TITLE
Non-breaking spaces not working correctly

### ISSUE THREAD

**Issue by akuma119** [hjson/hjson#102]
https://imgur.com/xGH9jZC

A coworker using sublime text uses **&nbsp** characters for spaces.
If I remove the **&nbsp** characters it works fine.
Shouldn't it work anyways?

**Comment by dqsully** [comment]
No. The only "whitespace" characters in Hjson are space (0x20), tab (0x09), new line/line feed (0x0A) and carriage return (0x0D). When you use the non-breaking space character Hjson will treat it like text. For example

**Comment by akuma119** [comment]
why? is this a design decision?

**Comment by sffc** [comment]
**&nbsp;** (U+00A0) is not normally treated as an ignorable whitespace character.  You can find a list of characters that Unicode considers ignorable in pattern syntax here:

http://unicode.org/L2/L2005/05012r-pattern.html

Unicode recommends effectively 9 ignorable characters:

```
0009..000D     # Cc   [5] <control-0009>..<control-000D>
0020           # Zs       SPACE
0085           # Cc       <control-0085>
200E..200F     # Cf   [2] LEFT-TO-RIGHT MARK..RIGHT-TO-LEFT MARK
2028           # Zl       LINE SEPARATOR
2029           # Zp       PARAGRAPH SEPARATOR
```

I think @dqsully's list of ASCII-range whitespace characters in common use is a reasonable set.
